whois-api/src/main/java/net/ripe/db/whois/api/rest/HttpsBasicAuthCustomizer.java

```diff
@@ -0,0 +1,90 @@
+package net.ripe.db.whois.api.rest;
+
+import com.google.common.net.HttpHeaders;
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletRequestWrapper;
+import jakarta.servlet.http.HttpServletResponse;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.jetty.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.codec.binary.Base64;
+import static jakarta.servlet.http.HttpServletRequest.BASIC_AUTH;
+
+@Component
+public class HttpsBasicAuthCustomizer implements Filter {
+
+    @Override
+    public void doFilter(final ServletRequest request, final ServletResponse response, final FilterChain chain) throws IOException, ServletException {
+
+        if(!canProceed(request)) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        final HttpServletRequest httpRequest = (HttpServletRequest) request;
+
+        if (RestServiceHelper.isHttpProtocol(httpRequest)){
+            final HttpServletResponse httpResponse = (HttpServletResponse)response;
+            httpResponse.sendError(HttpStatus.UPGRADE_REQUIRED_426, "HTTPS required for Basic authorization");
+            return;
+        }
+
+        chain.doFilter(new HttpBasicAuthRequestWrapper((HttpServletRequest) request), response);
+    }
+
+    private boolean canProceed(final ServletRequest request) {
+       if(! (request instanceof HttpServletRequest)) {
+           return false;
+       }
+
+       if(!RestServiceHelper.isBasicAuth((HttpServletRequest) request)) {
```

Comment:
here you can directly 
return !RestServiceHelper.isBasicAuth((HttpServletRequest) request)

and remove the return true; below, previous code will tackle this already